internal/expr/parser.go

```diff
@@ -611,20 +751,19 @@ func (p *Parser) parseTargetTypes() (types []valueAccessor, parentheses bool, ok
 // be followed by a name byte.
 func (p *Parser) parseOutputExpression() (*outputPart, bool, error) {
 	start := p.pos
+	cp := p.save()
 
 	// Case 1: There are no columns e.g. "&Person.*".
 	if targetType, ok, err := p.parseTargetType(); err != nil {
 		return nil, false, err
 	} else if ok {
 		return &outputPart{
 			sourceColumns: []columnAccessor{},
-			targetTypes:   []valueAccessor{targetType},
+			targetTypes:   []keyedAccesor{targetType},
 			raw:           p.input[start:p.pos],
 		}, true, nil
 	}
 
-	cp := p.save()
```

Comment:
Is there a good reason to move this to the start here? If `parseTargetType` is doing what we expect this should be safe.

Reply:
None, it was a remnant of the previous layout of the code. I left it there just for consistency of having the checkpoints at the top of the file, but I will leave it out of this PR.

Reply:
Cool ok. There was a discussion a while back about weather to keep the checkpoints consistent in each function (i.e. always start and end) or weather to do the save as late as possible. It was decided that we should generally do the most efficient option and leave the save as late as we can.